tests/test_bayes_search.py

```diff
@@ -1126,3 +1126,34 @@ def opt_func(run):
             num_iterations=5,
             atol=0.01,
         )
+
+
+def test_runs_bayes_impute_while_running():
+
+    sweep_config = {
+        "method": "bayes",
+        "metric": {
+            "name": "loss",
+            "goal": "minimize",
+            "impute": "worst",
+            "impute_while_running": True,
+        },
+        "early_terminate": {
+            "type": "hyperband",
+            "max_iter": 18,
+            "eta": 3,
+            "s": 2,
+        },
+        "parameters": {"a": {"values": [1, 2, 3]}},
+    }
+
+    run = next_run(sweep_config, [])
+    run.state = RunState.running
+    run.history = [{"loss": 10} for _ in range(4)]
+    run2 = next_run(sweep_config, [run])
+    run2.state = RunState.running
+    run2.history = [{"loss": 10 - i} for i in range(10)]
+    to_stop = stop_runs(sweep_config, [run, run2])
+
+    assert to_stop[0] is run
+    assert to_stop[0].early_terminate_info["bands"] == [2, 6]
```

Comment:
I think the test we want to write here is the following:

1. Take an existing bayes search test that checks if we converge to the best value of the metric using finished runs
2. Modify the test so that every time we add a new run, we do not set the previous run's state to finished but instead keep it running
3. Turn on running-run imputation with `best`, and ensure the best value of the metric of the running run is the same as what would have been used for the finished run in the original test
4. Check that the search converges to the same answer as the original test.

Reply:
okay I'm pretty sure the new version of the test does this, but definitely not positive. @dannygoldstein

Reply:
Well to be more accurate, the test I created looked visually effective and works for 2 python versions, but definitely looks weird for py36 and py39.... strange.

Reply:
what we want to do here is start with a simple bayes search test, like `test_squiggle_convergence_full`. then we want to modify the inner loop of `run_bayes_search` so that we dont set runs to finished when they are done, but instead keep them running (this line here): 

https://github.com/wandb/sweeps/pull/95/files#diff-b2272e756d704353b0bff5c05dd273880e2d5a0ea39e3dc35ef55f616e9eab76R44

Then we should alter the sweep config so that we enable `"impute_while_running"` (but not `"impute"`, lets keep that concern separate) 

And then we just want to check that we get the identical answer to the original `test_squiggle_convergence_full`.

Reply:
Okay, done. The test adds a good 5 minutes but goes through all the 5 parameters the original `test_squiggle_convergence_full` covers. I could delete that initial test actually, since it is strictly duplicated with the new test. Thoughts? @dannygoldstein